network/eos/eos_eapi.py

```diff
@@ -0,0 +1,273 @@
+            - Port on which the HTTP server will listen.
+        required: false
+        default: 80
+    https_port:
+        description:
+            - Port on which the HTTPS server will listen.
+        required: false
+        default: 443
+    local_http_port:
+        description:
+            - Port on which the local HTTP server will listen.
+        required: false
+        default: 8080
+    http:
+        description:
+            - Enable HTTP server access.
+        required: false
+        default: true
+    https:
+        description:
+            - Enable HTTPS server access.
+        required: false
+        default: true
+    local_http:
+        description:
+            - Enable local HTTP server access.
+        required: false
+        default: false
+    socket:
+        description:
+            - Enable Unix socket server access.
+        required: false
+        default: false
+"""
+
+EXAMPLES = """
+    - name: Enable EAPI access with default configuration
+      eos_eapi:
+          state: started
+          provider: {{ provider }}
+
+    - name: Enable EAPI with no HTTP, HTTPS at port 9443, local HTTP at port 80, and socket enabled
+      eos_eapi:
+          state: started
+          http: false
+          https_port: 9443
+          local_http: yes
+          local_http_port: 80
+          socket: yes
+          provider: {{ provider }}
+
+    - name: shutdown EAPI access
+      eos_eapi:
+          state: stopped
+          provider: {{ provider }}
+"""
+
+RETURN = """
+changed:
+    description:
+        - Indicates if commands were sent to the device.
+    returned: always
+    type: boolean
+    sample: false
+
+commands:
+    description:
+        - Set of commands to be executed on remote device
+    returned: always
+    type: list
+    sample: [
+        'config',
+        'management api http-commands',
+        'shutdown'
+    ]
+
+_config:
+    description:
+       - Configuration found on the device priot ro any commands being executed.
+    returned: always
+    type: object
+    sample: {...}
+"""
+
+
+def http_commands(protocol, port, enable, config):
+
+    started_config = config['{0}Server'.format(protocol)]
+    commands = []
+    changed = False
+
+    log(str(started_config))
+    log('\n')
+
+    if started_config.get('running'):
+        if not enable:
+            # turn off server
+            commands.append('no protocol {0}'.format(protocol))
+            changed = True
+        elif started_config.get('port') != port:
+            # update the port
+            commands.append('protocol {0} port {1}'.format(protocol, port))
+            changed = True
+    elif not started_config.get('runnng') and enable:
+        # turn on server
+        commands.append('protocol {0} port {1}'.format(protocol, port))
+        changed = True
+
+    return commands, changed
+
+
+def execute_commands(module, commands):
+
+    if not module.params.get('check_mode'):
+        module.execute(commands)
```

Comment:
this should be calling module.configure not module.execute and 'config' can be dropped from the commands list once done